packages/babylon/src/parser/statement.js

```diff
@@ -460,7 +460,8 @@ pp.parseBlockBody = function (node, allowDirectives, topLevel, end) {
     let stmt = this.parseStatement(true, topLevel);
 
     if (allowDirectives && !parsedNonDirective &&
-        stmt.type === "ExpressionStatement" && stmt.expression.type === "StringLiteral") {
+        stmt.type === "ExpressionStatement" && stmt.expression.type === "StringLiteral" &&
+        !stmt.expression.extra.parenthesized) {
```

Comment:
I assume there's no need to guard against non-existence of `extra` here (`(stmt.expression.extra && ! stmt.expression.extra.parenthesized)`).